projects.yaml

```diff
@@ -36,542 +36,673 @@ module:
         - records/daplink/interface.yaml
         - records/daplink/target_family.yaml
         - records/daplink/target_board.yaml
+
     # HICs
     hic_k20dx: &module_hic_k20dx
-        - records/rtos/rtos-cm3.yaml
         - records/hic_hal/k20dx.yaml
     hic_kl26z: &module_hic_kl26z
-        - records/rtos/rtos-cm0.yaml
         - records/hic_hal/kl26z.yaml
     hic_lpc11u35: &module_hic_lpc11u35
-        - records/rtos/rtos-cm0.yaml
         - records/hic_hal/lpc11u35.yaml
     hic_lpc4322: &module_hic_lpc4322
-        - records/rtos/rtos-cm3.yaml
         - records/hic_hal/lpc4322.yaml
     hic_sam3u2c: &module_hic_sam3u2c
-        - records/rtos/rtos-cm3.yaml
         - records/hic_hal/sam3u2c.yaml
     hic_stm32f103xb: &module_hic_stm32f103xb
-        - records/rtos/rtos-cm3.yaml
         - records/hic_hal/stm32f103xb.yaml
     hic_max32620: &module_hic_max32620
-        - records/rtos/rtos-cm4.yaml
         - records/hic_hal/max32620.yaml
     hic_max32625: &module_hic_max32625
-        - records/rtos/rtos-cm4.yaml
         - records/hic_hal/max32625.yaml
 
+    # RTOS types
+    rtx4_cm0: &module_rtx4_cm0
+        - records/rtos/rtos-rtx4-cm0.yaml
+    rtx4_cm3: &module_rtx4_cm3
+        - records/rtos/rtos-rtx4-cm3.yaml
+    rtx4_cm4: &module_rtx4_cm4
+        - records/rtos/rtos-rtx4-cm4.yaml
+    rtos_none: &module_rtos_none
+        - records/rtos/rtos-none.yaml
+
```

Comment:
Lets's simplify the changes here.
- Revert HIC yaml modules so they include the RTOS records.
- Remove RTOS yaml modules.
- For the bootloader projects, add the HIC yaml record plus RTOS-none yaml record directly.
- Add `records/rtos/rtos-none.yaml` directly into `module_bl`.

Example bootloader project:
```yaml
k20dx_bl:
  - *module_bl
  - records/hic_hal/k20dx.yaml
  - records/board/k20dx_bl.yaml
```

These changes will make it so the interface projects yaml don't have to change at all. All the differences are pushed onto the bootloader projects, which are few in number.